python/tank/platform/engine.py

```diff
@@ -1736,12 +1819,17 @@ def start_engine(engine_name, tk, context):
 
     # Instantiate the engine
     class_obj = loader.load_plugin(plugin_file, Engine)
-    obj = class_obj(tk, context, engine_name, env)
+    engine = class_obj(tk, context, engine_name, env)
 
     # register this engine as the current engine
-    set_current_engine(obj)
+    set_current_engine(engine)
 
-    return obj
+    # if the engine supports logging metrics, initialize the workers
+    metrics_dispatch_queue = MetricsDispatchQueue()
```

Comment:
I think i understand more now that I have seen the code further down! :)

This may be minutia, if you feel that's the case feel free to ignore it! :)
The fact that the queue is a global and singleton etc makes complete sense.
What confuses me is that the dispatcher is global. Maybe it's just me, but I think it would be elegant if there was a clearer separation between the production of metrics and consumption of metrics. So all the stuff that adds metrics stays the same, but in the engine it would look something like this:

```python
# in the engine code after post_app_init - all apps have now initialized, 
# the engine has reached a settled state. At this point we have already queued up a whole bunch
# of metrics. Now kick off the dispatch.

# (under the hood in the coonstructor, this will connect to the 
# one in-memory queue that exists as part of the singleton)
# maybe we can pass in a tk instance or engine instance if we want logging etc.
self._metrics_dispatcher = MetricsDispatcher(self)
# and kick of it's thread to start do work
self._metrics_dispatcher.start()
```

Then, in the engine tear down, we deallocate

```python
# gracefully stop the metrics dispatch thread.
self._metrics_dispatcher.stop()
```